src/components/__tests__/HamburgerMenu.test.jsx

```diff
@@ -0,0 +1,71 @@
+import i18next from 'i18next';
+import {fromJS} from 'immutable';
+import constant from 'lodash-es/constant';
+import React from 'react';
+import {Provider} from 'react-redux';
+import {act, create} from 'react-test-renderer';
+import configureStore from 'redux-mock-store';
+
+import {MenuItem} from '../TopBar/createMenu';
+import HamburgerMenu from '../TopBar/HamburgerMenu';
+
+jest.mock('i18next', () => {
+  return {
+    t: jest.fn(),
+    init: jest.fn(),
+  };
+});
+
+i18next.t.mockImplementation(constant('MOCKED_TRANSLATION_STRING'));
+
+const mockStore = configureStore([]);
+const store = mockStore(
+  fromJS({
+    ui: {
+      openTopBarMenu: 'hamburger',
+    },
+  }),
+);
+
+const DEFAULT_PROPS = {
+  hasInstructions: false,
+  isEditingInstructions: false,
+  isUserAuthenticated: false,
+  onAutoFormat: jest.fn(),
+  onStartEditingInstructions: jest.fn(),
+  onStartGithubLogIn: jest.fn(),
+};
+
+function buildComponent() {
+  return (
+    <Provider store={store}>
+      <HamburgerMenu {...DEFAULT_PROPS} />
+    </Provider>
+  );
+}
+
+function renderComponent(props = {}) {
+  let component;
+  act(() => {
+    component = create(buildComponent(props));
+  });
+  return component;
+}
+
+describe('hamburger menu', () => {
+  let menu;
+
+  beforeEach(() => {
+    menu = renderComponent();
+  });
+
+  it('calls the passed `onAutoFormat()` method when format code button is pressed', () => {
+    const firstMenuItem = menu.root.findAllByType(MenuItem)[0];
```

Comment:
Would it be fairly straightforward to scan for a menu item with the text label we expect? I could see this turning into a false failure if a future change introduced a menu item above format code that was always on, for instance.

Reply:
@outoftime This is a good callout. The issue is that the translation function `i18next.t()` is mocked and returns a fixed value. 

What I can do is amend the mock to return back the translation key and use that to select the menu item.

Ideally would be good to use `data-testid` attrs in combination with `@testing-library/react` or `enzyme` but that's for another PR.